src/qibocal/auto/task.py

```diff
@@ -223,19 +222,22 @@ def results(self, value):
 
     def dump(self):
         """Dump object to disk."""
-        if self.path is None:
-            raise ValueError("No known path where to dump execution results.")
-
-        self.path.mkdir(parents=True, exist_ok=True)
-        self.task.dump(self.path)
-        if self._data is not None:
-            self._data.save(self.path)
-        if self._results is not None:
-            self._results.save(self.path)
+        if self.path is not None:
```

Comment:
To make it simpler, just do the opposite
```suggestion
        if self.path is None:
            return None
```

I see you wanted to drop the error,  but not doing anything in a function (as in the case of an empty `else`) is fully equivalent to `return None`.
And you'll save an indentation level (in general, early returns are more readable).